UserTools/EventSelector/EventSelector.cpp

```diff
@@ -206,10 +218,20 @@ bool EventSelector::Execute(){
   bool passThroughGoingCut = true;
   m_data->Stores.at("RecoEvent")->Set("ThroughGoing",passThroughGoingCut);
 
-  std::vector<double> cluster_reco_pdg;
-  bool passRecoPDGCut = this->EventSelectionByRecoPDG(fRecoPDG, cluster_reco_pdg);
-  m_data->Stores.at("RecoEvent")->Set("RecoPDGVector",cluster_reco_pdg);
-  m_data->Stores.at("RecoEvent")->Set("PDG",fRecoPDG);
+  bool passRecoPDGCut = 0;
+  std::cout << "fRecoPDG: " << fRecoPDG << endl;
+  if (fRecoPDG != -1) {
+      std::vector<double> cluster_reco_pdg;
+      passRecoPDGCut = this->EventSelectionByRecoPDG(fRecoPDG, cluster_reco_pdg);
+      m_data->Stores.at("RecoEvent")->Set("RecoPDGVector", cluster_reco_pdg);
+      m_data->Stores.at("RecoEvent")->Set("PDG", fRecoPDG);
+  }
+
+  bool passExtendedCut = this->EventSelectionByTriggerExtended();
```

Comment:
nitpicking but this keeps coming up: there are only a very small number of obscure cases involving derived class pointers being dynamically cast to base class pointers where `this->` is required. Just call the function:
`passExtendedCut = EventSelectionByTriggerExtended();`